test/builder-api/src/packages.js

```diff
@@ -233,6 +269,24 @@ describe('Working with packages', function () {
         });
     });
 
+    // TED: Story time. This test is actually validating broken behavior. It's here to make sure we
+    // are consistently broken when it comes to package version sorting.
+    it('returns the latest release of a package with the specified odd name', function (done) {
+      request.get('/depot/pkgs/neurosis/testapp2/latest')
+        .type('application/json')
+        .accept('application/json')
+        .expect(200)
+        .end(function (err, res) {
+          expect(res.body.ident.origin).to.equal('neurosis');
+          expect(res.body.ident.name).to.equal('testapp2');
+          // This is what the expectation would be if our sorting logic worked in the core crate
```

Comment:
Hmm, it's not clear to me the sorting is broken... in these sort of non-standard versions, it seems correct to sort the package version "1.2.3-master" as greater (later) than "1.2.3-aaster".